test/python/synthesis/test_permutation_synthesis.py

```diff
@@ -40,6 +42,42 @@ def test_get_ordered_swap(self, width):
             self.assertTrue(np.array_equal(permutation, output))
             self.assertLess(len(swap_list), width)
 
+    @data(4, 5, 10, 15, 20)
+    def test_synth_permutation_basic(self, width):
+        """Test synth_permutation_basic function produces the correct
+        circuit."""
+        np.random.seed(1)
+        for _ in range(5):
+            pattern = np.random.permutation(width)
+            qc = synth_permutation_basic(pattern)
+
+            # Check that the synthesized circuit consists of SWAP gates only.
+            for instruction in qc.data:
+                self.assertEqual(instruction.operation.name, "swap")
+
+            # Construct a linear function from the synthesized circuit, and
+            # check that its permutation pattern matches the original pattern.
+            synthesized_pattern = LinearFunction(qc).permutation_pattern()
+            self.assertTrue(np.array_equal(synthesized_pattern, pattern))
+
+    @data(4, 5, 10, 15, 20)
+    def test_synth_permutation_acg(self, width):
```

Comment:
it's worth to add a test that the SWAP depth is 2, as well as to check that the number of SWAPs is the same as in the basic method

Reply:
Good point, adding this check.